R/viz.R

```diff
@@ -3045,8 +3066,396 @@ fancy_vplot = function(this_sample,
+  y = c(1:22)
+  yend = c(1:22)
+
+  #transform to data frame
+  segment_data = data.frame(chr, chr_start, chr_end, cent_start, cent_end, y, yend)
+
+  #load CN data
+  cn_states = get_sample_cn_segments(this_sample_id = this_sample, multiple_samples = FALSE, with_chr_prefix = FALSE, streamlined = FALSE)
+
+  #convert chr into y coordinates
+  cn_states$ycoord = cn_states$chrom
+
+  #paste chr in chromosomecolumn, if not there
+  if(!str_detect(cn_states$chrom, "chr")){
+    cn_states = mutate(cn_states, chrom = paste0("chr", chrom))
+  }
+
+  #convert data types
+  cols.int = c("start", "end", "ycoord")
+  cn_states[cols.int] = sapply(cn_states[cols.int], as.integer)
+  cn_states$chrom = as.factor(cn_states$chrom)
+  cn_states$CN = as.factor(cn_states$CN)
+
+  #subset on CN state
+  if("0" %in% levels(cn_states$CN)){
+    cn_0 = dplyr::filter(cn_states, CN == 0)
+  }
+  if("1" %in% levels(cn_states$CN)){
+    cn_1 = dplyr::filter(cn_states, CN == 1)
+  }
+  if("3" %in% levels(cn_states$CN)){
+    cn_3 = dplyr::filter(cn_states, CN == 3)
+  }
+  if("4" %in% levels(cn_states$CN)){
+    cn_4 = dplyr::filter(cn_states, CN == 4)
+  }
+  if("5" %in% levels(cn_states$CN)){
+    cn_5 = dplyr::filter(cn_states, CN == 5)
+  }
+  if("6+" %in% levels(cn_states$CN)){
+    cn_6 = dplyr::filter(cn_states, CN >= 6)
+  }
+
+  #load maf data
+  if(include_sv){
+    maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = coding_only, from_flatfile = from_flatfile, use_augmented_maf = use_augmented_maf)$maf
+
+    #transform maf data
+    maf_trans = dplyr::select(maf, Chromosome, Start_Position, End_Position, Variant_Type)
+
+    #calculate mid points of variants
+    maf_trans$mid = ((maf_trans$End_Position - maf_trans$Start_Position) / 2) + maf_trans$Start_Position
+
+    #convert chr into y coordinates
+    maf_trans$ystart = maf_trans$Chromosome
+    maf_trans$yend = maf_trans$Chromosome
+
+    #paste chr in maf, if not there
+    if(!str_detect(maf_trans$Chromosome, "chr")){
+      maf_trans = mutate(maf_trans, Chromosome = paste0("chr", Chromosome))
+    }
+
+    #convert data types
+    maf_trans$Start_Position = as.double(maf_trans$Start_Position)
+    maf_trans$End_Position = as.double(maf_trans$End_Position)
+    maf_trans$ystart = as.integer(maf_trans$ystart)
+    maf_trans$yend = as.integer(maf_trans$yend)
+
+    #subset on variant type
+    maf_del = dplyr::filter(maf_trans, Variant_Type == "DEL")
+    maf_ins = dplyr::filter(maf_trans, Variant_Type == "INS")
+
+    if(sv_count){
+      del_count = dplyr::filter(maf_trans, Variant_Type == "DEL") %>%
+        add_count(Chromosome) %>%
+        distinct(Chromosome, .keep_all = TRUE) %>%
+        dplyr::select(Chromosome, Variant_Type, yend, n)
+
+      ins_count = dplyr::filter(maf_trans, Variant_Type == "INS") %>%
+        add_count(Chromosome) %>%
+        distinct(Chromosome, .keep_all = TRUE) %>%
+        dplyr::select(Chromosome, Variant_Type, yend, n)
+    }
+  }
+
+  #plot
+  ggplot() +
```

Comment:
this function does not seem to return any object. Sometimes it makes it difficult if you want to store a plot in a variable and then combine multiple plots with something like ggpubr:: ggarrange when arranging a multiplot figure. Like the next function below `fancy_multisamp_ideogram` can we store the plot here in something like `p` and then return it?

Reply:
Absolutely!

Reply:
Also, this function should also be updated to use the newly added helper function to subset cn states for the sample.